### ISSUE TITLE
[JavaScript] Go to panel not working correctly with functions

### ISSUE THREAD

**Issue by ccampbell** [sublimehq/Packages#188]
Also related to changes in 409ccb469bca8bcc2a5c70c562d1db3428d3015f

Take this for example

```javascript
function SomeClass() {
    return {
        hello: function() {},
        goodBye: function() {}
    }
}
```

Go to panel shows:

![](https://devi.vimeocdn.com/craig/Nnwg)

It matches the functions but highlights all of them grouped together when you select one.

Previously (in a91eb66ef181399398950fe6470825b735905b85):

![](https://devi.vimeocdn.com/craig/qEfc)

**Comment by wbond** [comment]
Good catch!

**Comment by wbond** [comment]
This should be resolved by 412635a10fe39ae507656d6aee7e502180de2505
